File: 프로그래머스/lv1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> answer = new ArrayList<>();
        for (int[] cmd : commands) {
            int[] slicedArray = Arrays.copyOfRange(array, cmd[0]-1, cmd[1]);
            Arrays.sort(slicedArray);
            answer.add(slicedArray[cmd[2]-1]);
        }
        
        return answer.stream().mapToInt(x -> x.intValue()).toArray();
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (auto &cmd : commands) {
        vector<int> sub_array(array.begin() + cmd[0] - 1, array.begin() + cmd[1]);
        sort(sub_array.begin(), sub_array.end());
        answer.push_back(sub_array[cmd[2] - 1]);
    }
    
    return answer;
}